app/src/main/java/com/arctouch/codechallenge/view/home/HomeActivity.kt

```diff
@@ -0,0 +1,187 @@
+package com.arctouch.codechallenge.view.home
+
+import android.app.SearchManager
+import android.arch.lifecycle.Observer
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.databinding.DataBindingUtil
+import android.net.ConnectivityManager
+import android.os.Bundle
+import android.support.design.widget.Snackbar
+import android.support.v4.content.ContextCompat
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.RecyclerView
+import android.support.v7.widget.SearchView
+import android.util.Log
+import android.view.Menu
+import android.view.MenuItem
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.databinding.HomeActivityBinding
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.util.ConnectivityReceiver
+import com.arctouch.codechallenge.util.Tags
+import com.arctouch.codechallenge.util.getViewModel
+import com.arctouch.codechallenge.view.details.MovieDetailsFragment
+import com.arctouch.codechallenge.viewmodel.MoviesViewModel
+import com.arctouch.codechallenge.web.api.MovieRepository
+import com.arctouch.codechallenge.web.api.MovieService
+import com.arctouch.codechallenge.web.api.TmdbApi
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+
+
+class HomeActivity : AppCompatActivity(), HomeAdapter.ListListener, ConnectivityReceiver.ConnectivityReceiverListener {
+
+    private val mCompositeDisposable = CompositeDisposable()
+    private lateinit var mBinding: HomeActivityBinding
+    private lateinit var mViewModel: MoviesViewModel
+    private var mIsLoading = false
+    private val mConnectivityReceiver = ConnectivityReceiver(this)
+    private var snackBar: Snackbar? = null
+
+    companion object {
+        const val FRAGMENT_TAG = "details_fragment"
+    }
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        mViewModel = createViewModel()
+        configureDataBinding()
+        registerReceiver(mConnectivityReceiver, IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION))
+    }
+
+    private fun configureDataBinding() {
+        mBinding = DataBindingUtil.setContentView(this, R.layout.home_activity)
+        mBinding.lifecycleOwner = this
+
+        val adapter = HomeAdapter(emptyList<Movie>().toMutableList())
+        adapter.listener = this
+        mBinding.recyclerView.adapter = adapter
+
+        this.lifecycle.addObserver(mViewModel)
+        mBinding.viewModel = mViewModel
+
+        mBinding.recyclerView.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+            override fun onScrolled(recyclerView: RecyclerView?, dx: Int, dy: Int) {
+                super.onScrolled(recyclerView, dx, dy)
+                if (!recyclerView!!.canScrollVertically(1)
```

Comment:
Avoid to use `!!` and let in consideration to handle null-safety properly on Kotlin, like:

    recyclerView?.let { recyclerView ->
       if (!recyclerView.canScrollVertically(1)){
           // code goes here
       }
    }

Reply:
Thanks for the tip!!